### DISCUSSION TITLE
Troubleshooting Fundamentals

### DISCUSSION THREAD

**Discussion by kiot389** [Security-Onion-Solutions/securityonion#6988] [Unsupported Versions] [answered]
I just started using SO and its amazing stuff. I did a network install on Ubuntu. It went perfectly and I used it for a week. I'm good with Linux, and understand SO, but just starting using it. I ran into a services issue and need a place to start troubleshooting. Like, kernel.log, journald, dmsg. So-status shows Wait-Start for most services except krator, docker-registery, and aptcacherng. I'd like to know the fundamental places to start checking for issues. 

Where should a new user begin troubleshooting?

**Comment by abesinger** [comment]
Just a couple of tidbits of information:

- logs from SO components are generally in `/opt/so/log`, not in `/var/log` (but there _are_ things in `/var/log` worth checking).  Check the Salt logs for errors making updates/changes, sometimes Salt is the problem, not the particular service.
 
- `/sbin/so-*` are mostly (all?) bash scripts, sometimes reading them can help you understand how certain things are done, in addition to looking at the Salt configs under /opt/so/saltstack.

- you can find a number of posts here where the problem was firewall configuration.  Get familiar with `so-allow` and check whether it's put what you need in iptables.

**Comment by dougburks** [comment]
Also see:
https://docs.securityonion.net/en/2.3/help.html

**Comment by kiot389** [comment]
Perfect, this led me to find the details in the salt logs. Tricky commands. I started troubleshooting, but wanted to get back to using the system. So, I reinstalled Ubuntu, ran set up, and I was back to the login about thirty mins later.

**Comment by kiot389** [comment]
My issue turned out to be a typo and the hardware. I built the system but didn't tighten the network card. In the process of troubleshooting, I swapped to the nic's 2nd port, typo'd the static DNS. Fix that and the dashboard lit up.

Some standard things I checked were:
* packet collection. /nsm/suricata, /nsm/pcap. 
* made sure docker was up: docker logs so-suricata
* had to get introduced to salt.  sudo salt-call state.highstate
* https://docs.securityonion.net/en/2.3/search.html?q=salt&check_keywords=yes&area=default

**Comment by kiot389** [comment, marked as answer]
The details I was looking for are in the dashboard's Cheat Sheet.

![image](https://user-images.githubusercontent.com/36932098/152708274-b96a93b6-6f95-4849-bb4e-b041690b3bdc.png)

![image](https://user-images.githubusercontent.com/36932098/152708325-7e382162-c878-46b0-9e7c-e0ba1a3aaf63.png)

![image](https://user-images.githubusercontent.com/36932098/152708335-9b1c930b-0caa-42af-bc0f-a4604b01dfbf.png)
